benchmarks/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/ScopePassingTraceContextHookConfig.java

```diff
@@ -0,0 +1,133 @@
+		return new ScopePassingTraceContextHookRegisteringBeanDefinitionRegistryPostProcessor(context);
+	}
+
+}
+
+class ScopePassingTraceContextHookRegisteringBeanDefinitionRegistryPostProcessor extends HookRegisteringBeanDefinitionRegistryPostProcessor {
+
+	private static final Log log = LogFactory.getLog(ScopePassingTraceContextHookRegisteringBeanDefinitionRegistryPostProcessor.class);
+	private final ConfigurableApplicationContext context;
+
+	ScopePassingTraceContextHookRegisteringBeanDefinitionRegistryPostProcessor(ConfigurableApplicationContext context) {
+		super(context);
+		this.context = context;
+	}
+
+	@Override
+	void setupHooks( BeanFactory beanFactory) {
+		Hooks.onEachOperator(
+				TraceReactorAutoConfiguration.TraceReactorConfiguration.SLEUTH_TRACE_REACTOR_KEY,
+				scopePassingSpanOperator(this.context));
+		Schedulers.setFactory(factoryInstance(beanFactory));
+	}
+
+
+	private static <T> SpanSubscriptionProvider<T> scopePassingTraceContextSubscription(
+			ConfigurableApplicationContext beanFactory, Scannable scannable, CoreSubscriber<? super T> sub) {
+		return new SpanSubscriptionProvider<T>(
+				beanFactory,
+				sub,
+				sub.currentContext(),
+				scannable.name()) {
+			@Override
+			SpanSubscription newCoreSubscriber(Tracing tracing) {
+				return new ScopePassingTraceContextSubscriber(
+						sub,
+						sub != null ? sub.currentContext() : Context.empty(),
+						tracing);
+			}
+		};
+	}
+
+	private Schedulers.Factory factoryInstance(final BeanFactory beanFactory) {
+		return new Schedulers.Factory() {
+			@Override
+			public ScheduledExecutorService decorateExecutorService(String schedulerType,
+			                                                        Supplier<? extends ScheduledExecutorService> actual) {
+				return new TraceableScheduledExecutorService(beanFactory,
+						actual.get());
+			}
+		};
+	}
+
+
+	public static <T> Function<? super Publisher<T>, ? extends Publisher<T>> scopePassingSpanOperator(
+			ConfigurableApplicationContext beanFactory) {
+		return sourcePub -> {
+			// TODO: Remove this once Reactor 3.1.8 is released
+			//do the checks directly on actual original Publisher
+			if (sourcePub instanceof ConnectableFlux //Operators.lift can't handle that
+					|| sourcePub instanceof GroupedFlux  //Operators.lift can't handle that
+			) {
+				return sourcePub;
+			}
+			//no more POINTCUT_FILTER since mechanism is broken
+			Function<? super Publisher<T>, ? extends Publisher<T>> lift = Operators.lift((scannable, sub) -> {
+				//rest of the logic unchanged...
+				if (beanFactory.isActive()) {
+					if (log.isTraceEnabled()) {
+						log.trace("Spring Context already refreshed. Creating a scope " + "passing span subscriber with Reactor Context " + "[" + sub.currentContext() + "] and name [" + scannable.name() + "]");
+					}
+					return scopePassingTraceContextSubscription(beanFactory, scannable, sub).get();
```

Comment:
seems more efficient to `return new ScopePassingTraceContextSubscriber(...` vs making a provider and immediately calling get.